app/src/main/java/com/zeapo/pwdstore/sshkeygen/SshKeyGenFragment.kt

```diff
@@ -10,81 +10,128 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.view.inputmethod.InputMethodManager
-import android.widget.ArrayAdapter
-import android.widget.Button
-import android.widget.CheckBox
-import android.widget.EditText
-import android.widget.Spinner
+import android.widget.Toast
+import androidx.core.content.edit
 import androidx.core.content.getSystemService
 import androidx.fragment.app.Fragment
-import com.google.android.material.textfield.TextInputEditText
+import androidx.preference.PreferenceManager
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.jcraft.jsch.JSch
+import com.jcraft.jsch.KeyPair
 import com.zeapo.pwdstore.R
+import com.zeapo.pwdstore.databinding.FragmentSshKeygenBinding
+import java.io.File
+import java.io.FileOutputStream
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
-class SshKeyGenFragment : Fragment() {
+class SshKeyGenFragment : Fragment(), CoroutineScope {
 
-    private lateinit var checkBox: CheckBox
-    private lateinit var comment: EditText
-    private lateinit var generate: Button
-    private lateinit var passphrase: TextInputEditText
-    private lateinit var spinner: Spinner
-    private lateinit var activity: SshKeyGenActivity
+    override val coroutineContext get() = Dispatchers.IO + Job()
```

Comment:
Just keep the main context to Dispatchers.Main() maybe because we are in a fragment.